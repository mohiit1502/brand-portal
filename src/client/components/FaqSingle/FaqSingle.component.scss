.c-FaqSingle {
    border-bottom: 1px solid #CCCCCC;

    ol li {
        padding-left: 0.5rem;
    }

    ul {
        padding-left: 0.8rem;
        &.dashed {
            list-style-type: none;
            & > li {
                padding-left: 0;
                &:before {
                    content: "-";
                    margin-right: 1rem;
                    font-weight: bolder;
                }
            }
        }
    }
}
