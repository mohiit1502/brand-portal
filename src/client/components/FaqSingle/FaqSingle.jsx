/* eslint-disable filenames/match-regex */
/* eslint-disable no-use-before-define */
import React, {useEffect, useState} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {toggleImageViewer} from "../../actions/content/content-actions";
import Accordion from "./../Accordion";
import ContentRenderer from "../../utility/ContentRenderer";
import "./FaqSingle.component.scss";

const FaqSingle = ({data, expandPreState, toggleImageViewerDispatcher,goToContactUs}) => {

  const [expanded, setExpanded] = useState(expandPreState ? expandPreState : false);
  const [contentRenderer, setContentRenderer] = useState();

  useEffect(() => {
    setContentRenderer(new ContentRenderer(data, "c-FaqSingle__helpImage", toggleImageViewerDispatcher,goToContactUs));
  }, [data]);

  return (
    <div className="c-FaqSingle" id={data.id}>
      <Accordion data={data} expanded={expanded} setExpanded={setExpanded}>{contentRenderer && contentRenderer.generateContentDOM(expanded, expandPreState)}</Accordion>
    </div>
  );
};

FaqSingle.propTypes = {
  data: PropTypes.object,
  expandPreState: PropTypes.bool,
  toggleImageViewerDispatcher: PropTypes.func,
  goToContactUs: PropTypes.func
};

const mapDispatchToProps = {
  toggleImageViewerDispatcher: toggleImageViewer
}

export default connect(null, mapDispatchToProps)(FaqSingle);
