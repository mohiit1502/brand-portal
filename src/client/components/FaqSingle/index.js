import FaqSingle from './FaqSingle.jsx';

export default FaqSingle;