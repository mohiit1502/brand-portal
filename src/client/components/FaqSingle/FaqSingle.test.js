/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import FaqSingle from "./FaqSingle";

describe("FaqSingle", () => {
   it("renders without error", () => {

   });
});
