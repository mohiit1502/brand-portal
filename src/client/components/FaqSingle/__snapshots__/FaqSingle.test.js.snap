// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqSingle renders without error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MockNextContext
    pathname="/help"
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/help",
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/help",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/help",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <withRouter(Connect(FaqSingle))
          data={Object {}}
          expandPreState={true}
          goToContactUs={[Function]}
          history={Object {}}
          toggleImageViewerDispatcher={[Function]}
        >
          <Connect(FaqSingle)
            data={Object {}}
            expandPreState={true}
            goToContactUs={[Function]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "pathname": "/help",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/help",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/help",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            toggleImageViewerDispatcher={[Function]}
          >
            <FaqSingle
              data={Object {}}
              expandPreState={true}
              goToContactUs={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "pathname": "/help",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/help",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/help",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": false,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              toggleImageViewerDispatcher={[Function]}
            >
              <div
                className="c-FaqSingle"
              >
                <Accordion
                  data={Object {}}
                  expanded={true}
                  setExpanded={[Function]}
                >
                  <div
                    className="c-Accordion"
                  >
                    <div
                      className="c-Accordion__button-container"
                      data-test="accordion-buttonPanel"
                    >
                      <button
                        className="c-Accordion__button expanded"
                        data-test="accordion-button"
                        onClick={[Function]}
                      />
                    </div>
                    <div
                      className="c-Accordion__panel expanded"
                      style={
                        Object {
                          "maxHeight": "100%",
                        }
                      }
                    />
                  </div>
                </Accordion>
              </div>
            </FaqSingle>
          </Connect(FaqSingle)>
        </withRouter(Connect(FaqSingle))>
      </Router>
    </MemoryRouter>
  </MockNextContext>
</Provider>
`;
